lib/faces_detect.py

```diff
@@ -1,38 +1,20 @@
-import dlib
-import face_recognition
-import face_recognition_models
+from lib import FaceLandmarksExtractor
 
 def detect_faces(frame, model="hog"):
-    face_locations = face_recognition.face_locations(frame, model=model)
-    landmarks = _raw_face_landmarks(frame, face_locations)
-
-    for ((y, right, bottom, x), landmarks) in zip(face_locations, landmarks):
-        yield DetectedFace(frame[y: bottom, x: right], x, right - x, y, bottom - y, landmarks)
-
-# Copy/Paste (mostly) from private method in face_recognition
-predictor_68_point_model = face_recognition_models.pose_predictor_model_location()
-pose_predictor = dlib.shape_predictor(predictor_68_point_model)
-
-def _raw_face_landmarks(face_image, face_locations):
-    face_locations = [_css_to_rect(face_location) for face_location in face_locations]
-    return [pose_predictor(face_image, face_location) for face_location in face_locations]
-
-def _css_to_rect(css):
-    return dlib.rectangle(css[3], css[0], css[1], css[2])
-# end of Copy/Paste
+    fd = FaceLandmarksExtractor.extract (frame, True if model == "cnn" else False )
+    for face in fd:
+        x, y, right, bottom, landmarks = face[0][0], face[0][1], face[0][2], face[0][3], face[1]
+        yield DetectedFace(frame[y: bottom, x: right], x, right - x, y, bottom - y, landmarksXY=landmarks)
 
 class DetectedFace(object):
-    def __init__(self, image=None, x=None, w=None, y=None, h=None, landmarks=None, landmarksXY=None):
+    def __init__(self, image=None, x=None, w=None, y=None, h=None, landmarksXY=None):
         self.image = image
         self.x = x
         self.w = w
         self.y = y
         self.h = h
-        self.landmarks = landmarks
+        self.landmarks = None
```

Comment:
Is that var still needed ? if we set it to `None` I assume it is no more used, so maybe we can remove it totally?

Reply:
I dont know , if remove it, may be problems with existing alignments.json ?

Reply:
The raw landmarks aren't stored in the alignments file so it looks like `self.landmarks` was only used in the function `landmarksAsXY` below to format the tuple as `(p.x, p.y)`, since this landmark extractor already outputs a tuple with this shape it should be fine.